introduce the Ember initiative (#2485)
mainmatter/mainmatter.com: src/posts/2024-07-08-the-embroider-initiative-becomes-the-ember-initiative.md

---
title: "The Embroider Initiative Becomes the Ember Initiative"
authorHandle: marcoow
tags: [ember]
bio: "Marco Otte-Witte"
description:
  "Marco Otte-Witte introduces the Ember Initiative – a permanent effort to
  advance the Ember ecosystem, building on the success of the Embroider
  Initiative."
og:
  image: /assets/images/posts/2024-07-08-the-embroider-initiative-becomes-the-ember-initiative/og-image.png
tagline:
  "<p>The Embroider initiative was a big success for the Ember ecosystem. Having
  two engineers work full-time on Embroider for a year allowed us to reach a
  point where new Ember apps can now be built with Vite from the start. It might
  not be as easy for teams working on slightly older Ember versions, but the
  path to Vite is open. Most teams just need to update their apps to the latest

>   Ember version to follow it to the end.</p> <p>While the Embroider initiative

Reviewer: > version to follow it to the end

I don't know what this is supposed to mean 🤔 Maybe something like "Most teams just need to update their apps to the latest Ember version to try out Vite for themselves"

Author: > but the path to Vite is open. Most teams just need to update their apps to the latest Ember version to follow it to the end.

in "follow it to the end", the "it" refers to the "path to Vite"

Reviewer: I mean I don't doubt that it does 😂 It's just that it's not very clear (in my opinion) when reading the sentence. Other reviewers could have a different opinion so I'm not insisting that it is fixed or anything (if I could even do something like that 😂 )

Author: rephrased

  has come to a successful end, we don’t want to stop pushing the Ember
  ecosystem forward. The success of the initiative has shown that crowd-funded
  work on Ember is effective: companies see the reasons to invest, and we can
  reach our goals, which have real value for everyone in the ecosystem. That’s
  why we’re proposing to turn the Embroider initiative into a permanent effort –
  the <strong>Ember Initiative</strong>.</p>"
image: "/assets/images/ember/ember-consulting.jpeg"

> imageAlt: "A group of hamsters"

Reviewer: This is not true 😂 these are guinea pigs in the photo. I don't know if it's intended to be a placeholder, if it's a mistake, or a joke.

Author: Ha, we've been using this photo with that alt text on other pages forever. I'd say most people won't know and for the ones who know it's a joke then :)

---

## What is the Ember Initiative about?

Mainmatter will continue to assign a team to work on Ember and its ecosystem
full-time, indefinitely if we secure enough funding (more on that below). They
will address topics relevant to the Ember ecosystem and every company that uses
Ember. The team won’t focus on a single topic like they did with the Embroider
initiative. Instead, they will work on various issues, addressing whatever is
most pressing for the Ember ecosystem at any given time. To start, we’re
proposing 4 main topics:

### Polishing Embroider

While the Embroider initiative has been a great success and the path towards
Vite is wide open, there is some work left to do to polish the experience:

> - **Backwards compatibility:** Making Vite builds work with older Ember versions
>   will make it easier for many teams to switch. We suggest working on making
>   Embroider and Vite compatible with Ember 4.x.

Reviewer: I find the "we suggest" phrasing a bit strange here 🤔 I think we could go with something like: 

```suggestion
- **Backwards compatibility:** Making Vite builds work with older Ember versions
  will make it easier for many teams to switch. The current plan is to for Embroider
  with Vite to support as far back as Ember 4.x.
```

But I don't know if this suggestion is a bit clunky

Author: it's very clunky I'd say

Author: I change the sentence. We can't say _"The current plan"_ as the sponsors will get a say on the plan as well – we're suggesting a plan but it's not set in stone as we suggest it.

> - **Documentation:** While building Ember apps with Vite should be an
>   out-of-the-box experience, some documentation is needed to make this
>   accessible for the wider Ember community.

Reviewer: ```suggestion
- **Documentation:** While building Ember apps with Vite should be an
  out-of-the-box experience, some documentation  will need to be updated 
  before it can be the default experience for the wider Ember community.
```

> - **Opening upgrade paths for the most popular addons:** Ember apps rely heavily
>   on addons, but not all widely used ones work with Vite. We suggest adding
>   support for these addons or documenting alternatives.

Reviewer: I think this isn't a great proposition. Again similar to my above suggestion, I think it should be a plan or a proposal. And "these add-ons" is a bit noncommittal so how about something like: 

```suggestion
- **Opening upgrade paths for the most popular addons:** Ember apps rely heavily
  on addons, but not all widely used ones work with Vite. We propose that we add support
  or document alternatives for the top 100 most used addons in the Ember ecosystem.
```

Author: ```suggestion
- **Opening upgrade paths for the most popular addons:** Ember apps rely heavily
  on addons, but not all widely used ones work with Vite. We propose that we add
  support or document alternatives for the most used addons in the Ember ecosystem.
```

Reviewer: yep that works 👍

past few years. We suggest working on:

- **Ember Inspector:** The inspector needs some changes to work properly with
  Vite-built Ember apps. Also, its capabilities for modern Ember patterns like
  `<template>` tag components are limited. Fixing this will make developers more
  efficient.
- **Glint tooling and TS support:** We suggest our team works with Alex
  Matchneer, who has already begun improving Ember’s developer experience with
  TS and moving Glint to Volar for easier maintainability (see
  [Alex’s talk about the topic at Ember Europe](https://www.youtube.com/watch?v=6zy4nLHj83g)).

### Compiling Ember Apps to Web Components

Teams and companies are increasingly adopting polyglot architectures for their
frontends, using a mix of different frameworks. Using web components in Ember
apps is already straightforward. We suggest creating an easy way to compile
Ember apps themselves into web components so they can be used in other
applications. This will make it significantly easier to use (and continue using)
Ember in these contexts.

### Route and Router Manager APIs

Improving the router has been a long-standing goal in the Ember community, with

> little progress in recent years (the topic was recently removed from Polaris,

Reviewer: "the topic" hasn't been removed, it has been downsized. The thing that has been removed is that we're not doing the whole new router. I feel that we need to clarify here or people might have a backlash that might obscure the good things in this post

Author: rephrased
